code/modules/client/preferences.dm

```diff
@@ -113,7 +113,10 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	var/loaded_preferences_successfully = load_preferences()
 	if(loaded_preferences_successfully)
 		if(load_character())
-			return
+			// Sanitize languages.
+			for(var/datum/language/lang_path as anything in languages)
+				if(initial(lang_path.secret))
+					languages.Remove(lang_path)
```

Comment:
```suggestion
			// SKYRAT EDIT START - Sanitizing languages
			for(var/datum/language/lang_path as anything in languages)
				if(initial(lang_path.secret))
					languages.Remove(lang_path)
			// SKYRAT EDIT END
```

Reply:
Also, we should figure out something where it makes sure that the languages you've got in there are valid languages still, as I'm almost certain the current errors with the language menu bugging out for some people is due to some languages getting renamed at some point or the other.
An `initial(lang_path.name) == default_language_name` check could probably do the trick for that too.

Reply:
`.name` is not used anywhere in logic, so this is a moot point!

It's all done via paths which are completely immutable unless I'm missing something.